pkg/webhook/util/ip.go

```diff
@@ -0,0 +1,77 @@
+package util
+
+import (
+	"net"
+)
+
+// incrementIP increments the IP address by 1
+func incrementIP(ip net.IP) net.IP {
+	// Convert the IP to a byte slice and increment the last byte
+	ip = ip.To4()
+	for j := len(ip) - 1; j >= 0; j-- {
+		ip[j]++
+		if ip[j] != 0 {
+			break
+		}
+	}
+	return ip
+}
+
+func GetUsableIPAddressesCount(includeRange string, excludeRange []string) (count int, err error) {
+	var includeRangeList []string
+	includeRangeList = append(includeRangeList, includeRange)
+
+	includeIPAddrList, err := getIPAddressesFromSubnet(includeRangeList, true)
+	if err != nil {
+		return count, err
+	}
+
+	excludeIPAddrList, err := getIPAddressesFromSubnet(excludeRange, false)
+	if err != nil {
+		return count, err
+	}
+
+	for includeIP := range includeIPAddrList {
+		if _, exists := excludeIPAddrList[includeIP]; !exists {
+			count++
+		}
+	}
+
+	return count, nil
+}
+
+func getIPAddressesFromSubnet(ipNetSubnets []string, include bool) (ipAddrList map[string]struct{}, err error) {
+	ipAddrList = make(map[string]struct{})
+
+	for _, ipNetSubnet := range ipNetSubnets {
+		_, network, err := net.ParseCIDR(ipNetSubnet)
+		if err != nil {
+			return ipAddrList, err
+		}
+
+		// Get broadcast address (last address in the subnet)
+		broadcast := getBroadcastAddress(network)
```

Comment:
Do we have to do this ourselves or can user include that in the exclude list, and make this a documentation item?

Reply:
users just specify subnet/IPNet in include, and wherabouts just calcuate the range excluding network and broadcast ip
for usable ip address.

It will not be correct for the user to specify this in exclude list.

The logic/functions added in pkg/webhook/util/ip.go is similar to what wherabouts CNI does it internally to know the ip addresses  that could be assigned to LH pods.